### PR TITLE
Don't map P-DEP SIMD12 local vars to SIMD16 on x64

### LINKED ISSUE #12950: AccessViolationException with SIMD type fields in struct when optimizations enabled
Executing [this](https://github.com/RossNordby/scratchpad/blob/d039476eb76f37c90e9c208585432e9e98b713b0/Repro/Repro/Program.cs) code:
```cs
using System.Numerics;

namespace Repro
{
    class Program
    {
        struct BoundingBoxTest
        {
            public Vector3 Min;
            public Vector3 Max;

            public override int GetHashCode()
            {
                return Min.GetHashCode() + Max.GetHashCode();
            }
        }

        public static void Test()
        {
            var box = new BoundingBoxTest();
            box.Min = Vector3.Min(box.Min, box.Min);
            var hmm = box.GetHashCode();
        }

        static void Main(string[] args)
        {
            var someMemory = new int[1];
            var someMoreMemory = new int[1];
            Test();
            someMoreMemory[someMemory[0]] = 0;
        }
    }
}

```
results in an AccessViolationException when compiled with optimizations enabled. It reproduces in every preview I've tried (from 2.0.0-preview2-002093-00 to 2.0.0-preview3-25518-01).

I didn't dig deep enough to identify a specific cause, but I did note that doing any the following seems to avoid the access violation (possibly only by chance):
1) Removing the BoundingBoxTest.GetHashCode override, 
2) Using StructLayout on the BoundingBoxTest to specify a size above the default of 24, and
3) Using non-SIMD types.

### PR DESCRIPTION
**Pull request by AndyAyersMS**
P-DEP local vars are logically independent locals, but physically embeded
in some structure with fixed layout. So they cannot be made larger.

We already had safeguards for ths in place for x86 so extend these to kick
in for x64 too.

Also update Lowering's checker to account for the fact that some SIMD12s
can persist in x64.

Fixes #12950.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@CarolEidt @BruceForstall PTAL
cc @dotnet/jit-contrib 

We should consider taking this for 2.0.

File: src/jit/lower.cpp [hunk 1]
```diff
@@ -4775,9 +4775,10 @@ void Lowering::CheckCall(GenTreeCall* call)
 //                      after lowering.
 //
 // Arguments:
+//   compiler - the compiler context.
 //   node - the node to check.
 //
-void Lowering::CheckNode(GenTree* node)
+void Lowering::CheckNode(Compiler* compiler, GenTree* node)
```
Comment (CarolEidt) [inline src/jit/lower.cpp]: Another alternative, a little cleaner, but slightly less efficient I suppose, would be to make this an instance method of `Lowering`, which has a `Compiler` member.

**Comment by BruceForstall** [comment]
I assume no diffs?

**Comment by AndyAyersMS** [comment]
@BruceForstall I doubt we'll see diffs but I'll double-check.

**Comment by AndyAyersMS** [comment]
Arm testing failure unrelated and I'm going to ignore it.

Desktop SPMI had 5 methods with diffs, all from the CQ perf RayTracer example, and all looked like examples where we were generating bad code. In the few I looked at in detail the overly wide writes would spill into other vectors so the corruption they caused might not have been easily observable.

**Pull request merged**